### ISSUE TITLE
CLI Handling

### ISSUE THREAD

**Issue by lixitrixi** [conjure-cp/conjure-oxide#103]
To improve the usability of Conjure Oxide as a command-line tool, Nik and I have brainstormed adding a few standard Unix CLI patterns:

1. `conjure solve` by default should accept problem input directly through stdin
The user enters Essence and submits with `Ctrl+D`
2. A source file can be given through an optional "path" positional argument
3. The format of this input (AST JSON, Essence) can be specified through an optional argument (default: Essence)

In short: a Unix filter

It's worth saying that this will of course be a future issue; just jotting down ideas!

**Comment by ChrisJefferson** [comment]
The main problem with conjure solve accepting via stdin is there is usually there are two inputs to conjure, an 'essence' file and a 'param' file -- so there would have to be some way of passing both files by stdin.

This could definately be handled of course, one option (which would work well for conjure oxide) would be to accept a json file, which could include include both the param and problem easily. Of course, this would be less useful for "general users".

**Comment by gskorokhod** [comment]
Duplicate of #199
